engine/classes/ElggEntity.php

```diff
@@ -1433,6 +1504,35 @@ protected function update() {
 			'time_updated' => $time,
 			'guid' => $guid,
 		]);
+		if ($ret === false) {
+			return false;
+		}
+
+		$this->attributes['time_updated'] = $time;
+
+		// Update secondary table
+		$attrs = $this->getSecondaryTableColumns();
+
+		$sets = array_map(function ($attr) {
+			return "$attr = :$attr";
+		}, $attrs);
+		$sets = implode(', ', $sets);
+
+		foreach ($attrs as $attr) {
+			$params[":$attr"] = ($attr === 'url') ? '' : (string) $this->attributes[$attr];
+		}
+		$params[':guid'] = $this->guid;
+
+		$db = $this->getDatabase();
+		$query = "
+			UPDATE {$db->prefix}{$this->type}s_entity
+			SET $sets
+			WHERE guid = :guid
+		";
+
+		if ($db->updateData($query, false, $params) === false) {
+			return false;
+		}
 
 		elgg_trigger_after_event('update', $this->type, $this);
```

Comment:
Nice side effect of this: the "update" event is now called after the full attributes are updated.